### PR TITLE
Improve authn/authz error log when query failed.

### PR DESCRIPTION
**Pull request by zhongwencool**
1. Add more detail log when authn/authz query error(locate the problem faster).
2. Don't log error when authn_redis query result is empty.
3. Always debug when tracing.

### CODE REVIEW

**Review by HJianBo — DISMISSED** [review]

File: apps/emqx/src/emqx_trace/emqx_trace_formatter.erl [hunk 1]
```diff
@@ -33,22 +34,36 @@ format(#{level := trace, event := Event, meta := Meta, msg := Msg},
...
+    case Meta2 =:= #{} of
         true -> [Packet, Payload];
-        false -> [Packet, ", ", map_to_iolist(Meta1), Payload]
+        false -> [Packet, ", ", map_to_iolist(Meta2), Payload]
     end.
 
+enrich(_, "") -> "";
+enrich(Key, Bin) -> [Key, Bin].
+
+format_meta_(Meta, Encode) ->
```
Comment (zmstone) [inline apps/emqx/src/emqx_trace/emqx_trace_formatter.erl:54]: nit: is elvis OK with this ?

maybe `do_format_meta`

File: apps/emqx_authn/src/simple_authn/emqx_authn_jwt.erl [hunk 2]
```diff
@@ -272,7 +273,7 @@ verify(JWS, [JWK | More], VerifyClaims) ->
             verify(JWS, More, VerifyClaims)
     catch
         _:_Reason:_Stacktrace ->
```
Comment (zmstone) [inline apps/emqx_authn/src/simple_authn/emqx_authn_jwt.erl:275]: nit: delete the `_Stacktrace` part

File: apps/emqx/include/logger.hrl [hunk 3]
```diff
@@ -76,14 +76,17 @@
 -define(TRACE_FILTER, emqx_trace_filter).
 
 %% Only evaluate when necessary
+%% Always debug the trace events.
 -define(TRACE(Event, Msg, Meta),
```
Comment (zmstone) [inline apps/emqx/include/logger.hrl:80]:
```suggestion
-define(TRACE(Tag, Msg, Meta),
```

File: apps/emqx/src/emqx_trace/emqx_trace_formatter.erl [hunk 4]
```diff
@@ -33,22 +34,36 @@ format(#{level := trace, event := Event, meta := Meta, msg := Msg},
...
+    Payload = enrich(", payload: ", Payload0),
+    Meta2 = maps:without([msg, clientid, peername, packet, payload], Meta1),
+    case Meta2 =:= #{} of
         true -> [Packet, Payload];
-        false -> [Packet, ", ", map_to_iolist(Meta1), Payload]
+        false -> [Packet, ", ", map_to_iolist(Meta2), Payload]
     end.
 
+enrich(_, "") -> "";
+enrich(Key, Bin) -> [Key, Bin].
```
Comment (zmstone) [inline apps/emqx/src/emqx_trace/emqx_trace_formatter.erl:52]:
```suggestion
enrich(Key, IoData) -> [Key, IoData].
```

File: apps/emqx/include/logger.hrl [hunk 5]
```diff
@@ -76,14 +76,17 @@
...
     begin
     case persistent_term:get(?TRACE_FILTER, undefined) of
         undefined -> ok;
         [] -> ok;
         List ->
            emqx_trace:log(List, Event, Msg, Meta)
```
Comment (zmstone) [inline apps/emqx/include/logger.hrl:86]:
```suggestion
           emqx_trace:log(List, Msg, Meta#{trace_tag => Event})
```

File: apps/emqx/include/logger.hrl [hunk 6]
```diff
@@ -76,14 +76,17 @@
...
-    end
+    end,
+    ?SLOG(debug, (emqx_trace_formatter:format_meta(Meta))#{msg => Msg, trace_event => Event},
+        #{trace_filter => ignore})
```
Comment (zmstone) [inline apps/emqx/include/logger.hrl:89]: Suggestion: add a `trace_tag` field to `Meta` for both trace and log.
trace handler should only match logs with `trace_tag` field,
but do not inverse-match `trace_filter := ignore`.


```suggestion
    ?SLOG(debug, (emqx_trace_formatter:format_meta(Meta))#{msg => Msg, trace_tag => Event}, #{})
```

**Review by zmstone — CHANGES_REQUESTED** [review]

File: apps/emqx/include/logger.hrl (hunk 6 above)
Comment (zmstone) [reply apps/emqx/include/logger.hrl]: I mean add it in this TRACE macro, but not where this macro is used.

Comment (zmstone) [reply apps/emqx/include/logger.hrl]: Sorry, trace_tag is in use and for both trace and log

maybe a is_trace=>true for traces?

**Review by zmstone — APPROVED** [review]

**Pull request merged**